crates/router/src/core/payments.rs

```diff
@@ -1029,7 +1030,11 @@ where
         .to_validate_request()?
         .validate_request(&req, &merchant_account)?;
 
-    let payment_id = id_type::GlobalPaymentId::generate(state.conf.cell_information.id.clone());
+    // If payment_id is not present in request, generate new payment_id
+    let payment_id = match payment_id {
+        Some(id) => id,
+        None => id_type::GlobalPaymentId::generate(state.conf.cell_information.id.clone()),
```

Comment:
when we get None here?

Reply:
We get `None` in `Create Intent`. We get `Some` in `Get Intent` and `Update Intent`